When i try to compile libopenshot on windows 10, i am getting following error

```
-----------------------------------------------------------------
          Welcome to the OpenShot Build System!

CMake will now check libopenshot's build dependencies and inform
you of any missing files or other issues.

For more information, please visit <http://www.openshot.org/>.
-----------------------------------------------------------------

Generating build files for OpenShot with CMake 3.16.0-rc4
  Building libopenshot (version 0.2.3)
  SO/API/ABI Version: 17

-- Could NOT find ImageMagick (missing: ImageMagick_Magick++_LIBRARY ImageMagick_MagickWand_LIBRARY ImageMagick_MagickCore_LIBRARY)
-- Could NOT find RESVG, using Qt SVG parsing instead (missing: RESVG_LIBRARY RESVG_INCLUDE_DIR)
-- Looking for system JsonCpp
-- Could NOT find jsoncpp (missing: JSONCPP_INCLUDE_DIR JSONCPP_LIBRARY)
-- Using embedded JsonCpp (not found or USE_SYSTEM_JSONCPP disabled)
-- FFmpeg - postproc not found.
-- FFmpeg - swresample not found.
PYTHON_MODULE_PATH: C:/Program Files (x86)/libopenshot/Lib/site-packages
-- Could NOT find Ruby (missing: RUBY_EXECUTABLE RUBY_INCLUDE_DIR RUBY_LIBRARY)
-- Doxygen found, documentation target enabled

To compile documentation in doc/html, run: 'make doc'
-- Could NOT find ImageMagick (missing: ImageMagick_Magick++_LIBRARY ImageMagick_MagickWand_LIBRARY ImageMagick_MagickCore_LIBRARY)
-- Could NOT find RESVG, using Qt SVG parsing instead (missing: RESVG_LIBRARY RESVG_INCLUDE_DIR)
-- Tests enabled, test executable will be built as tests/openshot-test
-- Cmake 3.11+ detected, enabling 'test' target

To run unit tests, use: 'make os_test' or 'make test'
-- Displaying feature summary

Build configuration:
-- The following features have been enabled:

 * JsonCpp (embedded), JsonCpp will be compiled from the bundled sources

-- The following OPTIONAL packages have been found:

 * PythonInterp (required version >= 3)
 * PythonLibs (required version >= 3)
 * Doxygen
 * PkgConfig

-- The following REQUIRED packages have been found:

 * Qt5Widgets
 * Qt5Core
 * Qt5Gui
 * Qt5Network (required version >= 5.5.1)
 * Qt5Multimedia
 * Qt5MultimediaWidgets
 * Qt5
 * FFmpeg
 * Threads
 * OpenMP
 * ZeroMQ
 * SWIG (required version >= 3.0)
 * UnitTest++
 * OpenShotAudio (required version >= 0.1.8)

-- The following features have been disabled:

 * IWYU (include-what-you-use), Scan all source files with 'iwyu'

-- The following OPTIONAL packages have not been found:

 * JsonCpp
 * cppzmq
 * Ruby
 * ImageMagick
 * RESVG

-- Configuring done
-- Generating done
-- Build files have been written to: C:/libopenshot/build
[  1%] Automatic MOC for target openshot
[  1%] Built target openshot_autogen
Scanning dependencies of target openshot
[  2%] Building CXX object src/CMakeFiles/openshot.dir/openshot_autogen/mocs_compilation.cpp.obj
[  3%] Building CXX object src/CMakeFiles/openshot.dir/AudioBufferSource.cpp.obj
[  4%] Building CXX object src/CMakeFiles/openshot.dir/AudioReaderSource.cpp.obj
[  5%] Building CXX object src/CMakeFiles/openshot.dir/AudioResampler.cpp.obj
[  6%] Building CXX object src/CMakeFiles/openshot.dir/CacheBase.cpp.obj
[  7%] Building CXX object src/CMakeFiles/openshot.dir/CacheDisk.cpp.obj
[  8%] Building CXX object src/CMakeFiles/openshot.dir/CacheMemory.cpp.obj
[  9%] Building CXX object src/CMakeFiles/openshot.dir/ChunkReader.cpp.obj
[ 10%] Building CXX object src/CMakeFiles/openshot.dir/ChunkWriter.cpp.obj
[ 11%] Building CXX object src/CMakeFiles/openshot.dir/Color.cpp.obj
[ 12%] Building CXX object src/CMakeFiles/openshot.dir/Clip.cpp.obj
[ 13%] Building CXX object src/CMakeFiles/openshot.dir/ClipBase.cpp.obj
[ 14%] Building CXX object src/CMakeFiles/openshot.dir/Coordinate.cpp.obj
[ 15%] Building CXX object src/CMakeFiles/openshot.dir/CrashHandler.cpp.obj
[ 16%] Building CXX object src/CMakeFiles/openshot.dir/DummyReader.cpp.obj
[ 17%] Building CXX object src/CMakeFiles/openshot.dir/ReaderBase.cpp.obj
[ 18%] Building CXX object src/CMakeFiles/openshot.dir/RendererBase.cpp.obj
[ 19%] Building CXX object src/CMakeFiles/openshot.dir/WriterBase.cpp.obj
[ 20%] Building CXX object src/CMakeFiles/openshot.dir/EffectBase.cpp.obj
[ 21%] Building CXX object src/CMakeFiles/openshot.dir/EffectInfo.cpp.obj
[ 22%] Building CXX object src/CMakeFiles/openshot.dir/FFmpegReader.cpp.obj
C:\libopenshot\src\FFmpegReader.cpp:41:17: note: #pragma message: You are compiling only with software decode
 #pragma message "You are compiling only with software decode"
                 ^
[ 23%] Building CXX object src/CMakeFiles/openshot.dir/FFmpegWriter.cpp.obj
C:\libopenshot\src\FFmpegWriter.cpp:41:17: note: #pragma message: You are compiling only with software encode
 #pragma message "You are compiling only with software encode"
                 ^
C:\libopenshot\src\FFmpegWriter.cpp: In member function 'void openshot::FFmpegWriter::SetOption(openshot::StreamType, std::string, std::string)':
C:\libopenshot\src\FFmpegWriter.cpp:426:11: error: 'AV_CODEC_ID_H265' was not declared in this scope
      case AV_CODEC_ID_H265 :
           ^
C:\libopenshot\src\FFmpegWriter.cpp:485:11: error: 'AV_CODEC_ID_H265' was not declared in this scope
      case AV_CODEC_ID_H265 :
           ^
C:\libopenshot\src\FFmpegWriter.cpp: In member function 'AVStream* openshot::FFmpegWriter::add_video_stream()':
C:\libopenshot\src\FFmpegWriter.cpp:1165:9: error: 'AV_CODEC_ID_H265' was not declared in this scope
    case AV_CODEC_ID_H265 :
         ^
C:\libopenshot\src\FFmpegWriter.cpp:1204:5: error: 'AVCodecContext' has no member named 'framerate'
  c->framerate = av_inv_q(c->time_base);
     ^
src\CMakeFiles\openshot.dir\build.make:356: recipe for target 'src/CMakeFiles/openshot.dir/FFmpegWriter.cpp.obj' failed
mingw32-make[2]: *** [src/CMakeFiles/openshot.dir/FFmpegWriter.cpp.obj] Error 1
CMakeFiles\Makefile2:270: recipe for target 'src/CMakeFiles/openshot.dir/all' failed
mingw32-make[1]: *** [src/CMakeFiles/openshot.dir/all] Error 2
makefile:150: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

--- a/src/FFmpegWriter.cpp
+++ b/src/FFmpegWriter.cpp
@@ -423,7 +423,7 @@ void FFmpegWriter::SetOption(StreamType stream, std::string name, std::string va
 							av_opt_set(c->priv_data, "preset", "veryslow", 0);
 						}
 						break;
-					case AV_CODEC_ID_H265 :
+					case AV_CODEC_ID_HEVC :
 						av_opt_set_int(c->priv_data, "qp", std::min(std::stoi(value), 51), 0); // 0-51
 						if (std::stoi(value) == 0) {
 							av_opt_set(c->priv_data, "preset", "veryslow", 0);
@@ -482,7 +482,7 @@ void FFmpegWriter::SetOption(StreamType stream, std::string name, std::string va
 							av_opt_set(c->priv_data, "preset", "veryslow", 0);
 						}
 						break;
-					case AV_CODEC_ID_H265 :
+					case AV_CODEC_ID_HEVC :
 						av_opt_set_int(c->priv_data, "crf", std::min(std::stoi(value), 51), 0); // 0-51
 						if (std::stoi(value) == 0) {
 							av_opt_set(c->priv_data, "preset", "veryslow", 0);
@@ -1162,7 +1162,7 @@ AVStream *FFmpegWriter::add_video_stream() {
 			case AV_CODEC_ID_AV1 :
 #endif
 			case AV_CODEC_ID_VP9 :
-			case AV_CODEC_ID_H265 :
+			case AV_CODEC_ID_HEVC :
 #endif
 			case AV_CODEC_ID_VP8 :
 			case AV_CODEC_ID_H264 :
@@ -1200,7 +1200,8 @@ AVStream *FFmpegWriter::add_video_stream() {
 	 identically 1. */
 	c->time_base.num = info.video_timebase.num;
 	c->time_base.den = info.video_timebase.den;
-#if LIBAVFORMAT_VERSION_MAJOR >= 56
+// AVCodecContext->framerate was added in FFmpeg 2.2
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 26, 0)
 	c->framerate = av_inv_q(c->time_base);
 #endif
 	st->avg_frame_rate = av_inv_q(c->time_base);

